### PR TITLE
Tidy up a bunch of components + fix drop caps

### PR DESCRIPTION
**Pull request by alexwlchan**
This is a precursor to #7595 and part of #6927; mostly moving components out of common and into the appropriate app.

This includes a fix for drop caps, which have been broken for a while. 🙈

### CODE REVIEW

File: content/webapp/components/HTMLSerializers/HTMLSerializers.tsx [hunk 1]
```diff
@@ -13,7 +13,15 @@ export const dropCapSerializer: HTMLSerializer<ReactElement> = (
...
-  if (type === Elements.paragraph && key === '0' && children[0] !== undefined) {
+  // Note: `key.toString() === '0'` might look strange, because isn't `key`
+  // already a string?
+  //
+  // For some reason, we're getting passed `key` as a number here, so we
+  // need this or drop caps don't work correctly.
+  //
+  // TODO: Work out why this isn't being caught by the type checker elsewhere,
+  // and fix it properly.
+  if (type === Elements.paragraph && (key === '0' || key.toString() === '0') && children[0] !== undefined) {
```
Comment (davidpmccormick) [inline content/webapp/components/HTMLSerializers/HTMLSerializers.tsx:24]: I'd be surprised if this didn't need a bit of a lint :trollface:

**Review by davidpmccormick — APPROVED** [review]

File: content/webapp/components/HTMLSerializers/HTMLSerializers.tsx (hunk 1 above)
Comment (davidpmccormick) [reply content/webapp/components/HTMLSerializers/HTMLSerializers.tsx:24]: I guess the `key === '0'` is surplus to requirements here also

Comment (alexwlchan) [reply content/webapp/components/HTMLSerializers/HTMLSerializers.tsx]: Good shout!

Comment (alexwlchan) [reply content/webapp/components/HTMLSerializers/HTMLSerializers.tsx]: How about now?

**Pull request merged**